src/EditorFeatures/CSharpTest/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs

```diff
@@ -0,0 +1,130 @@
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.ConvertSwitchStatementToExpression;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.ConvertSwitchStatementToExpression
+{
+    public partial class ConvertSwitchStatementToExpressionTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    {
+        internal override (DiagnosticAnalyzer, CodeFixProvider) CreateDiagnosticProviderAndFixer(Workspace workspace)
+            => (new ConvertSwitchStatementToExpressionDiagnosticAnalyzer(), new ConvertSwitchStatementToExpressionCodeFixProvider());
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertSwitchStatementToExpression)]
+        public async Task TestReturn()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Program
+{
+    int M(int i)
+    {
+        [||]switch (i)
+        {
+            case 1:
+                return 4;
+            case 2:
+                return 5;
+            case 3:
+                return 6;
+        }
+    }
+}",
+@"class Program
+{
+    int M(int i)
+    {
+        return i switch
+        {
+            1 => 4,
+            2 => 5,
+            3 => 6
+        };
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertSwitchStatementToExpression)]
+        public async Task TestSingleAssignment()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Program
+{
+    void M(int i)
+    {
+        int j;
+        [||]switch (i)
+        {
+            case 1:
+                j = 4;
+                break;
+            case 2:
+                j = 5;
+                break;
+            case 3:
+                j = 6;
+                break;
+        }
+    }
+}",
+@"class Program
+{
+    void M(int i)
+    {
+        int j;
+        j = i switch
+        {
+            1 => 4,
+            2 => 5,
+            3 => 6
+        };
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertSwitchStatementToExpression)]
+        public async Task TestMultiSingleAssignment()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Program
+{
+    void M(int i)
+    {
+        int j, k;
+        [||]switch (i)
+        {
+            case 1:
+                j = 4;
+                k = 5;
+                break;
+            case 2:
+                j = 6;
+                k = 7;
+                break;
+            case 3:
+                j = 8;
+                k = 9;
+                break;
+        }
+    }
+}",
+@"class Program
+{
+    void M(int i)
+    {
+        int j, k;
+        (j, k) = i switch
```

Comment:
cute :)

Reply:
would be cuter if we get https://github.com/dotnet/roslyn/pull/29645 for switch expressions as well.

Reply:
note: this is at the cuteness level that i'm concerned about it being rolled into the same option.   I can see people being fine with teh single-assignment/return forms.  But the tuple-assignment form seems like a bit of stretch.  I think this should be discussed in design meeting.  if they're ok with this, great!

Reply:
> note: this is specifically because this is an analyzer. If this were a refactoring, i wouldn't care.
> 
> That said, i imagine the number of cases this will find in a codebase is actually pretty small... so maybe it's NBD if it ends up behaving this way. So i'm actually retracting my exception here.

I agree. Since it's a cause of a few complexities in the analyzer (as I dig through the details), I think we can table that for later and add if ever requested.